File: LeetCode/203. Remove Linked List Elements/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode pre = dummy, cur = head;
        while (cur != null) {
            if (pre.next.val != val) {
                pre = pre.next;
                cur = pre.next;
            } else {
                pre.next = cur.next;
                cur = pre.next;
            }
        }
        return dummy.next;
    }
}

File: LeetCode/203. Remove Linked List Elements/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next, cur, last = head, dummy, head
        while last:
            cur.next = last
            if last.val != val:
                cur = last
            last = last.next
        cur.next = last
        return dummy.next

